### ISSUE TITLE
tf.BoostedTree or GBDTMO

### ISSUE THREAD

**Issue by samanemami** [zzd1992/GBDTMO-EX#4]
I am confused about the loss function that is used in the [GBDTMO](https://github.com/samanemami/GBDTMO).
First, I assumed that you used, the same as XGBoost, and then I found that you implemented it over the LightGBM function.
Also, I have another question about this model; What is the difference between the [GBDTMO](https://arxiv.org/abs/1909.04373) model and the [TFBT](https://ieeexplore.ieee.org/abstract/document/8257910) model from TensorFlow?

Also, would you please help with the number of trees parameter in the model. I am not sure which one is related to boosting epochs?

Thank you in advance for your consideration.

**Comment by zzd1992** [comment]
When you build a GBDTMO instance, there is no number of trees parameter.
This parameter is initialized when you call `train` method.
For example,
```python
booster = GBDTMulti(LIB, out_dim=out_dim, params=params)
booster.set_data((x_train, y_train), (x_valid, y_valid))
booster.train(num_boosters)
```

And thanks for reminding. I haven't noticed TFBT before.
I read that paper and I think the differences are:

1) TFBT is based on XGBoost but my work is based on LightGBM (multi-output histogram, best first leaf grow).
2) My work supports sparse predictions.
3) TFBT supports layer by layer boosting which seems to speed up the training. I think this is an interesting idea.

For information of parameters, you can read `docs` of GBDTMO project.
